sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -73,6 +73,21 @@ grammar SqlBase;
       return false;
     }
   }
+
+ /**
+   * This method will be called when the character stream ends and try to find out the
+   * unclosed bracketed comment.
+   * If the next character is -1, it means the end of the entire character stream match,
+   * and we throw exception to prevent executing the query.
+   */
+  public void end() {
+    int nextChar = _input.LA(1);
+    if (nextChar == -1) {
+      int pos = _input.index();
+      String str = _input.getText(new Interval(0, pos));
+      throw new RuntimeException("Unclosed bracketed comment: " + str + ", position: " + pos);
```

Comment:
since the sql string can be multi-line, we can make the error message more readable
```
Unclosed bracketed comment started at position $pos:
$str
```

Reply:
can we get the start position instead of the end position?

Reply:
And we can throw ParserException? and put the error in `QueryParsingErrors`